src/transformers/models/ict/modeling_ict.py

```diff
@@ -0,0 +1,904 @@
+    config_class = ICTGuidedUpsamplerConfig
+    # base_model_prefix = "ict"
+    # main_input_name = "pixel_values"
+    supports_gradient_checkpointing = True
+    _no_split_modules = []
+
+    def _init_weights(self, module: Union[nn.Linear, nn.Embedding, nn.LayerNorm]) -> None:
+        """Initialize the weights"""
+        pass
+
+    def _set_gradient_checkpointing(self, module, value: bool = False) -> None:
+        if isinstance(module, ICTGuidedUpsampler):
+            module.gradient_checkpointing = value
+
+
+ICT_GUIDED_UP_SAMPLER_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`ICTGuidedUpsamplerConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+ICT_GUIDED_UP_SAMPLER_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See [`ICTImageProcessor.__call__`]
+            for details.
+
+        head_mask (`torch.FloatTensor` of shape `(num_heads,)` or `(num_layers, num_heads)`, *optional*):
+            Mask to nullify selected heads of the self-attention modules. Mask values selected in `[0, 1]`:
+
+            - 1 indicates the head is **not masked**,
+            - 0 indicates the head is **masked**.
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        interpolate_pos_encoding (`bool`, *optional*):
+            Whether to interpolate the pre-trained position encodings.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+class ICTGuidedUpsampler(nn.Module):
+    def __init__(self, config: ICTGuidedUpsamplerConfig):
+        super().__init__(config)
+
+        generator = InpaintGenerator(config.residual_blocks)
+        discriminator = Discriminator(in_channels=3, use_sigmoid=config.gan_loss != "hinge")
+
+        # TODO How to load weights for G and D?
+        # if len(config.GPU) > 1:
+        #     generator = nn.DataParallel(generator, config.GPU)
+        #     discriminator = nn.DataParallel(discriminator , config.GPU)
+
```

Comment:
I wasn't sure how to initialize weights for the generator and discriminator like it [does in the original repo here](https://github.com/raywzy/ICT/blob/main/Guided_Upsample/src/models.py#L17-L18). 

I uploaded the weights to [sheonhan/ict-imagenet-32](https://huggingface.co/sheonhan/ict-imagenet-32/tree/main), so I assume I need to download from the hub using their specific names but wasn't sure whether I need to rely on `init_weight` function of a superclass.

Reply:
We don't need to load the model weights to initialize a model but the implementation of ICT makes it a bit more confusing than it is. The generator and discriminator models are initialized over [here](https://github.com/raywzy/ICT/blob/main/Guided_Upsample/src/models.py#L62) (lines 62-69)

The `BaseNetwork.load() method` is executed after this and it is optional, it loads pretrained weights and overwrites the parameters of the randomly initialized generator and discriminator.

You won't need to load any weights to initialize the HF ICTModel except the clustering file (can be replaced with a random vector of shape 512 x 3).